### PR TITLE
add region into address adhesion

### PR DESCRIPTION
**Pull request by Raulnet**
add region into address adhesion

### CODE REVIEW

File: src/Entity/NullablePostAddress.php [hunk 1]
```diff
@@ -88,19 +88,26 @@ class NullablePostAddress implements AddressInterface, GeocodableInterface, GeoP
      */
     private $longitude;
 
-    private function __construct(string $country, string $postalCode, ?string $cityName, ?string $street, float $latitude = null, $longitude = null)
+    /**
+     * @ORM\Column(nullable=true)
+     */
+    private $region;
+
+    private function __construct(string $country, string $postalCode, ?string $cityName, ?string $street, float $latitude = null, $longitude = null, ?string $region = null)
```
Comment (JeremieSamson) [inline src/Entity/NullablePostAddress.php]: multiline plz

File: src/Entity/NullablePostAddress.php [hunk 2]
```diff
@@ -208,7 +215,7 @@ public function equals(self $other): bool
             && mb_strtolower($this->cityName) === mb_strtolower($other->getCityName())
             && mb_strtolower($this->postalCode) === mb_strtolower($other->getPostalCode())
             && mb_strtolower($this->country) === mb_strtolower($other->getCountry())
-        ;
+            ;
```
Comment (JeremieSamson) [inline src/Entity/NullablePostAddress.php]: wasn't it well placed ?

File: src/Entity/PostAddress.php [hunk 3]
```diff
@@ -230,10 +244,10 @@ public function getGeocodableAddress(): string
...
     {
         return mb_strtolower($this->address) === mb_strtolower($other->getAddress())
-               && mb_strtolower($this->cityName) === mb_strtolower($other->getCityName())
-               && mb_strtolower($this->postalCode) === mb_strtolower($other->getPostalCode())
-               && mb_strtolower($this->country) === mb_strtolower($other->getCountry())
-        ;
+            && mb_strtolower($this->cityName) === mb_strtolower($other->getCityName())
+            && mb_strtolower($this->postalCode) === mb_strtolower($other->getPostalCode())
+            && mb_strtolower($this->country) === mb_strtolower($other->getCountry())
+            ;
```
Comment (JeremieSamson) [inline src/Entity/PostAddress.php]: same ?

File: app/migrations/Version20190123111436.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+final class Version20190123111436 extends AbstractMigration
+{
+    public function up(Schema $schema): void
+    {
+        $this->addSql('ALTER TABLE citizen_projects ADD address_region VARCHAR(255) DEFAULT NULL');
+        $this->addSql('ALTER TABLE adherents ADD address_region VARCHAR(255) DEFAULT NULL');
+        $this->addSql('ALTER TABLE committees ADD address_region VARCHAR(255) DEFAULT NULL');
+        $this->addSql('ALTER TABLE events ADD address_region VARCHAR(255) DEFAULT NULL');
+        $this->addSql('ALTER TABLE donations ADD address_region VARCHAR(255) DEFAULT NULL');
```
Comment (Remg) [inline app/migrations/Version20190123111436.php]: are these columns needed ?

File: tests/Entity/PostAddressTest.php [hunk 5]
```diff
@@ -24,7 +24,7 @@ public function testCreateFullFrenchAddress()
 
     public function testCreateFullForeignAddress()
     {
-        $address = PostAddress::createForeignAddress('US', '20500', 'Washington, DC', '1600 Pennsylvania Ave NW', 48.123456, 5.987654);
+        $address = PostAddress::createForeignAddress('US', '20500', 'Washington, DC', '1600 Pennsylvania Ave NW', null, 48.123456, 5.987654);
```
Comment (JeremieSamson) [inline tests/Entity/PostAddressTest.php]: multiline please :)

File: src/Entity/NullablePostAddress.php (hunk 1 above)
Comment (Raulnet) [reply src/Entity/NullablePostAddress.php]: fixed

File: src/Entity/PostAddress.php (hunk 3 above)
Comment (Raulnet) [reply src/Entity/PostAddress.php]: fixed

File: tests/Entity/PostAddressTest.php (hunk 5 above)
Comment (Raulnet) [reply tests/Entity/PostAddressTest.php]: fixed;

File: src/Entity/NullablePostAddress.php (hunk 2 above)
Comment (Raulnet) [reply src/Entity/NullablePostAddress.php]: fixed

**Pull request merged**